File: Daily_Codes_Answers/day-16/70_Climbing_Stairs.py
def factorial(s):
    num=1
    for i in range(s,0,-1):
        num*=i
    return num

def climbStairs(n):
    step =1
    temp =n
    for i in range(1,(temp//2)+1):
        num = factorial(n-1)/(factorial(i)*factorial(n-1-i))
        step+=num
        n-=1
    return int(step)

n1 = 5
n2 = 6

print("-- test case 1 --")
print(f"Input : {n1}\nOutput : {climbStairs(n1)}")
print("-- test case 2 --")
print(f"Input : {n2}\nOutput : {climbStairs(n2)}")

File: Daily_Codes_Answers/day-16/70_Climbing_Stairs.js
const factorial = (s)=>{
    let num=1;
    for(let i=s ; i>=1 ;i--){
        num*=i;
    }
    return num;
}

const climbStairs = (n)=>{
    let step =1;
    let temp =n;
    for(let i = 1 ; i<=temp/2 ; i++){
        let num = factorial(n-1)/(factorial(i)*factorial(n-1-i));
        step+=num;
        n--;
    }
    return step;
}

let n1 = 5;
let n2 = 6;

console.log("-- test case 1 --");
console.log("Input : "+n1+"\nOutput : "+climbStairs(n1));
console.log("-- test case 2 --");
console.log("Input : "+n2+"\nOutput : "+climbStairs(n2));